### ISSUE TITLE
NullReferenceException in monthCalendar after setting MinDate

### ISSUE THREAD

**Issue by kirsan31** [dotnet/winforms#3034]
* .NET Core Version: 3.1.2

* Have you experienced this same bug with .NET Framework?: **No**

**Problem description:**
Set `MinDate` (`monthCalendar1.MinDate = DateTime.Today;`) , then select some date in calendar - 
NullReferenceException occur:
<details><summary>NullReferenceException </summary>

```cs
   at System.Windows.Forms.MonthCalendar.MonthCalendarAccessibleObject.GetCalendarCell(Int32 calendarIndex, AccessibleObject parentAccessibleObject, Int32 columnIndex)
   at System.Windows.Forms.MonthCalendar.MonthCalendarAccessibleObject.GetCalendarChildAccessibleObject(Int32 calendarIndex, CalendarChildType calendarChildType, AccessibleObject parentAccessibleObject, Int32 index)
   at System.Windows.Forms.MonthCalendar.MonthCalendarAccessibleObject.GetCalendarChildAccessibleObject(DateTime selectionStart, DateTime selectionEnd)
   at System.Windows.Forms.MonthCalendar.WmDateChanged(Message& m)
   at System.Windows.Forms.MonthCalendar.WmReflectCommand(Message& m)
   at System.Windows.Forms.MonthCalendar.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.DebuggableCallback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)
   at System.Windows.Forms.UnsafeNativeMethods.SendMessage(HandleRef hWnd, Int32 msg, IntPtr wParam, IntPtr lParam)
   at System.Windows.Forms.Control.SendMessage(Int32 msg, IntPtr wparam, IntPtr lparam)
   at System.Windows.Forms.Control.ReflectMessage(IntPtr hWnd, Message& m)
   at System.Windows.Forms.Control.WmNotify(Message& m)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.ScrollableControl.WndProc(Message& m)
   at System.Windows.Forms.ContainerControl.WndProc(Message& m)
   at System.Windows.Forms.Form.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.DebuggableCallback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)
   at System.Windows.Forms.UnsafeNativeMethods.CallWindowProc(IntPtr wndProc, IntPtr hWnd, Int32 msg, IntPtr wParam, IntPtr lParam)
   at System.Windows.Forms.NativeWindow.DefWndProc(Message& m)
   at System.Windows.Forms.Control.WmMouseDown(Message& m, MouseButtons button, Int32 clicks)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.DebuggableCallback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)
   at System.Windows.Forms.UnsafeNativeMethods.DispatchMessageW(MSG& msg)
   at System.Windows.Forms.Application.ComponentManager.System.Windows.Forms.UnsafeNativeMethods.IMsoComponentManager.FPushMessageLoop(IntPtr dwComponentID, Int32 reason, Int32 pvLoopData)
   at System.Windows.Forms.Application.ThreadContext.RunMessageLoopInner(Int32 reason, ApplicationContext context)
   at System.Windows.Forms.Application.ThreadContext.RunMessageLoop(Int32 reason, ApplicationContext context)
   at System.Windows.Forms.Application.Run(Form mainForm)
   at WinFormsCoreTest1.Program.Main() in d:\save\projects\core tests\WinFormsCoreTest3\Program.cs:line 24
```

</details>

![mc](https://user-images.githubusercontent.com/17767561/78277443-33b19980-751d-11ea-9c8b-2941f19da488.gif)

**Expected behavior:**
No exception.

Windows 7 x64.

**Minimal repro:**
[WinFormsCoreTest4.zip](https://github.com/dotnet/winforms/files/4422459/WinFormsCoreTest4.zip)

<details><summary>P.s.</summary>

We had grate plans about migration our apps to core. A lot of work were done in research, converting old project to new sdk format and then adopting to core... And after brief view on the result (not even testing!!!) I have found **4** problems that make core variant of WinForms unusable :( At least, until version 5.0 I think: ((((

</details>

**Comment by RussKie** [comment]
@merriemcgaw we'll have to patch this.

> And after brief view on the result (not even testing!!!) I have found **4** problems that make core variant of WinForms unusable

That's not good. Please raise these problems here as separate issues, and we'll have a look at them.
3.1 is a LTS release, so we'll consider patching broken blocking behaviour.

**Comment by kirsan31** [comment]
@RussKie Already done, 3 other issues:
https://github.com/dotnet/winforms/issues/3022
https://github.com/dotnet/winforms/issues/3032
https://github.com/dotnet/winforms/issues/3033

**Comment by RussKie** [comment]
Could you please add the stack trace to the top post?

**Comment by kirsan31** [comment]
> Could you please add the stack trace to the top post?

Done, under `NullReferenceException` tag in [1 post](https://github.com/dotnet/winforms/issues/3034#issue-592796235).

**Comment by Vino-Wang** [comment]
This is a duplicate issue to https://github.com/dotnet/winforms/issues/2475, @vladimir-krestov is working on it.

**Comment by kirsan31** [comment]
> @RussKie This is a duplicate issue to #2475, @vladimir-krestov is working on it.

 Oh, sorry - my bad :-(

**Comment by John-Qiao** [comment]
@vladimir-krestov the issue in [#2475](https://github.com/dotnet/winforms/issues/2475) cannot reproduce in the latest .NET Core 5.0 SDK, but this current issue still can reproduce in latest .NET Core 5.0 SDK, please confirm if they are the same issue.

**Comment by vladimir-krestov** [comment]
@John-Qiao, 
They are the same issues. Please look at the comment - "[This happens if the MinDate and MaxDate are set.](https://github.com/dotnet/winforms/issues/2475#issue-533694143)"
I couldn't reproduce NRE in a case if a min/max date **isn't set**.
Maybe you checked this case?

**Comment by John-Qiao** [comment]
@vladimir-krestov  yes, you're right, it occurs when a min/max date is set.

**Comment by kirsan31** [comment]
Fixed in 3.1.**4**.
![mc](https://user-images.githubusercontent.com/17767561/81780833-d8050380-94ff-11ea-84da-4aafa26ba000.gif)

**Comment by RussKie** [comment]
Thank you for the follow up!
